osquery/tables/system/linux/pci_devices.cpp

```diff
@@ -26,38 +31,162 @@ const std::string kPCIKeyVendor = "ID_VENDOR_FROM_DATABASE";
 const std::string kPCIKeyModel = "ID_MODEL_FROM_DATABASE";
 const std::string kPCIKeyID = "PCI_ID";
 const std::string kPCIKeyDriver = "DRIVER";
+const std::string kPCISubsysID = "PCI_SUBSYS_ID";
+
+const std::string kPciIdsPath = "/usr/share/misc/pci.ids";
+
+PciDB::PciDB(std::istream& db_filestream) {
+  // pci.ids keep track of subsystem information of vendor and models
+  // sequentially so we keep track of what the current vendor and models are.
+  std::string cur_vendor, cur_model, line;
+  while (std::getline(db_filestream, line)) {
+    if (line.size() < 7 || line.at(0) == '\n' || line.at(0) == '#') {
```

Comment:
I would suggest
```
line = line.substr(0, line.find_first_of('#'))
boost:trim_right(line);
if(line.size() < 7){
   continue;
}
```
looks more readable and takes more corner cases into the consideration.